pkg/apiserver/api_installer.go

```diff
@@ -321,10 +321,12 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 
 			// list or post across namespace.
 			// TODO: more strongly type whether a resource allows these actions on "all namespaces" (bulk delete)
-			namer = scopeNaming{scope, a.group.Linker, gpath.Join(a.prefix, itemPath), true}
-			actions = appendIf(actions, action{"LIST", resource, params, namer}, isLister)
-			actions = appendIf(actions, action{"POST", resource, params, namer}, isCreater)
-			actions = appendIf(actions, action{"WATCHLIST", "watch/" + resource, params, namer}, allowWatchList)
+			if !hasSubresource {
+				namer = scopeNaming{scope, a.group.Linker, gpath.Join(a.prefix, itemPath), true}
+				actions = appendIf(actions, action{"LIST", resource, params, namer}, isLister)
+				actions = appendIf(actions, action{"POST", resource, params, namer}, isCreater)
```

Comment:
Not sure if the scheduler has been switched - there was a commented out block when I put pods/binding in

> On Jun 18, 2015, at 12:02 AM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/apiserver/api_installer.go:
> 
> > @@ -321,10 +321,12 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
> >  
> >  			// list or post across namespace.
> >  			// TODO: more strongly type whether a resource allows these actions on "all namespaces" (bulk delete)
> > -			namer = scopeNaming{scope, a.group.Linker, gpath.Join(a.prefix, itemPath), true}
> > -			actions = appendIf(actions, action{"LIST", resource, params, namer}, isLister)
> > -			actions = appendIf(actions, action{"POST", resource, params, namer}, isCreater)
> > -			actions = appendIf(actions, action{"WATCHLIST", "watch/" + resource, params, namer}, allowWatchList)
> > +			if !hasSubresource {
> > +				namer = scopeNaming{scope, a.group.Linker, gpath.Join(a.prefix, itemPath), true}
> > +				actions = appendIf(actions, action{"LIST", resource, params, namer}, isLister)
> > +				actions = appendIf(actions, action{"POST", resource, params, namer}, isCreater)
> Oh, this is the deprecated top-level /binding API. Can't we nuke that entirely by now?
> https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/master/master.go#L467
> 
> —
> Reply to this email directly or view it on GitHub.
>